lib/auth/apiserver.go

```diff
@@ -245,10 +246,30 @@ func NewAPIServer(config *APIConfig) (http.Handler, error) {
 // HandlerWithAuthFunc is http handler with passed auth context
 type HandlerWithAuthFunc func(auth ClientI, w http.ResponseWriter, r *http.Request, p httprouter.Params, version string) (interface{}, error)
 
+func WithClientAddr(r *http.Request) (*http.Request, error) {
+	ctx := r.Context()
+	if ctx.Value(utils2.ContextClientAddr) != nil {
+		return r, nil
+	}
+	addr, err := utils.ParseHostPortAddr(r.RemoteAddr, 0)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	tcpAddr := &net.TCPAddr{
+		IP:   net.ParseIP(addr.Host()),
+		Port: addr.Port(0),
+	}
+	return r.WithContext(context.WithValue(ctx, utils2.ContextClientAddr, tcpAddr)), nil
```

Comment:
I like that we are using the same context key in the gRPC and HTTP side, this seems like it'll make it clearer how this is best accessed when the Client's IP is required.